src/charm.py

```diff
@@ -815,6 +815,8 @@ def _install_snap_packages(self, packages: List[str]) -> None:
                     snap_package.ensure(
                         snap.SnapState.Latest, channel=snap_channel, revision=snap_revision
                     )
+                    # snaps will auto refresh so it is necessary to hold the current revision
+                    snap_package.hold()
```

Comment:
**suggestion:** Deindent this one level. You probably want to hold regardless, not just if it's not present.